### PR TITLE
Properly detect labels for virtual machines in GCE

### PR DESCRIPTION
**Pull request by liu-samuel**
Properly detect labels for virtual machines in GCE

### CODE REVIEW

**Review by agrare — APPROVED** [review]

File: app/models/manageiq/providers/google/inventory/parser.rb [hunk 1]
```diff
@@ -206,6 +206,23 @@ def instances
...
+    end
+  end
+
+  def vm_and_template_labels(resource, tags)
+    tags.each do |tag|
+      persister
+        .vm_and_template_labels
+        .find_or_build_by(
+          :resource => resource,
+          :name     => tag.first,
```
Comment (agrare) [inline app/models/manageiq/providers/google/inventory/parser.rb:219]: Hm actually just noticed this .first and .second.

Is tags an array of arrays like `[[key1, value1], [key2, value2]]`?  Or is it a hash?

Comment (liu-samuel) [reply app/models/manageiq/providers/google/inventory/parser.rb:219]: (byebug) tag
[:"https-server", ""]
(byebug) tags
{:"http-server"=>"", :"https-server"=>""}

this is what I see in my breakpoint, so tags is a hash, and I guess when it iterates through the hash it converts it into an array?

Comment (agrare) [reply app/models/manageiq/providers/google/inventory/parser.rb:219]: Yeah okay so it is a hash with `{key => value}`

Can you change the loop to `tags.each do |tag_name, tag_value|`

Comment (liu-samuel) [reply app/models/manageiq/providers/google/inventory/parser.rb]: Sure, I pushed it up. Is that what you're looking for?

Comment (agrare) [reply app/models/manageiq/providers/google/inventory/parser.rb]: Yes looks good now thanks

**Comment by Fryguy** [comment]
Backported to `radjabov` in commit 65f345c8579fc11df2d668744bf12692da54564d.

```text
commit 65f345c8579fc11df2d668744bf12692da54564d
Author: Adam Grare <adam@grare.com>
Date:   Tue Sep 17 14:21:28 2024 -0400

    Merge pull request #269 from liu-samuel/google-compute-engine-labels
    
    Properly detect labels for virtual machines in GCE
    
    (cherry picked from commit ff8431d6b1e50c2f3cd4fa847046d1e2ebaab84b)
```

**Pull request merged**